lib/fs.js

```diff
@@ -648,6 +654,14 @@ fs.readSync = function(fd, buffer, offset, length, position) {
     offset = 0;
   }
 
+  if (length === 0) {
```

Comment:
The check here means that even if the file doesn't exist/inaccessible, if the length is zero, we will return valid values.